src/Symfony/Component/DependencyInjection/Definition.php

```diff
@@ -876,13 +876,19 @@ public function setBindings(array $bindings)
     /**
      * Add an error that occurred when building this Definition.
      *
-     * @param string $error
+     * @param string|\Closure|self $error
      *
      * @return $this
      */
     public function addError($error)
     {
-        $this->errors[] = $error;
+        if ($error instanceof self) {
+            foreach ($error->errors as $error) {
+                $this->errors[] = $error;
+            }
```

Comment:
Maybe just: `array_merge($this->errors, $error->errors);`

Reply:
done